integration/tests/basic/erc/test_EIP1559.py

```diff
@@ -0,0 +1,1317 @@
+        result = int(tx_receipt.logs[0].topics[1].hex(), 16)
+        assert result == 1
+
+    @pytest.mark.parametrize(*NEGATIVE_PARAMETERS)
+    def test_transfer_negative(
+            self,
+            max_priority_fee_per_gas,
+            max_fee_per_gas,
+            expected_exception,
+            exception_message_regex,
+    ):
+        sender = self.accounts[0]
+        recipient = self.web3_client.create_account()
+        value = 1
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id="auto",
+            from_=sender.address,
+            to=recipient.address,
+            value=value,
+            nonce="auto",
+            gas="auto",
+            max_priority_fee_per_gas=max_priority_fee_per_gas,
+            max_fee_per_gas=max_fee_per_gas,
+            data=None,
+            access_list=None,
+        )
+
+        with pytest.raises(expected_exception=expected_exception, match=exception_message_regex):
+            self.web3_client.send_transaction(
+                account=sender,
+                transaction=tx_params,
+                timeout=TX_TIMEOUT,
+            )
+
+    @pytest.mark.parametrize(*NEGATIVE_PARAMETERS)
+    def test_deploy_negative(
+            self,
+            max_priority_fee_per_gas,
+            max_fee_per_gas,
+            expected_exception,
+            exception_message_regex,
+    ):
+        account = self.accounts[0]
+
+        contract_iface = helpers.get_contract_interface(
+            contract="common/Common.sol",
+            version="0.8.12",
+            contract_name="Common",
+        )
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id="auto",
+            from_=account.address,
+            to=None,
+            value=0,
+            nonce="auto",
+            gas="auto",
+            max_priority_fee_per_gas=max_priority_fee_per_gas,
+            max_fee_per_gas=max_fee_per_gas,
+            data=contract_iface["bin"],
+            access_list=None,
+        )
+
+        with pytest.raises(expected_exception=expected_exception, match=exception_message_regex):
+            self.web3_client.send_transaction(
+                account=account,
+                transaction=tx_params,
+                timeout=TX_TIMEOUT,
+            )
+
+    def test_insufficient_funds(
+            self,
+    ):
+        sender = self.accounts[0]
+        balance = self.web3_client.get_balance(sender.address)
+        recipient = self.web3_client.create_account()
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id="auto",
+            from_=sender.address,
+            to=recipient.address,
+            value=balance,
+            nonce="auto",
+            gas="auto",
+            max_priority_fee_per_gas="auto",
+            max_fee_per_gas="auto",
+            data=None,
+            access_list=None,
+        )
+
+        error_msg_regex = r"{'code': -32000, 'message': '.+'}"
```

Comment:
can we check the error message too?

Reply:
**local**
`{ValueError}ValueError({'code': -32000, 'message': 'insufficient funds for transfer: address 0xc7780BD1f47450e42cB5a2e8b84FEc282E6cC022 have 200000000000000000000 want 200686152623224622240'})`

**geth**
`{ValueError}ValueError({'code': -32000, 'message': 'insufficient funds for gas * price + value: balance 200000000000000000000, tx cost 200000000570335325000, overshot 570335325000'})`

the first part of the message matches. Is it OK like this?
```suggestion
        error_msg_regex = r"{'code': -32000, 'message': 'insufficient funds for.+'}"
```